dissect/target/plugins/browsers/firefox.py

```diff
@@ -114,3 +139,80 @@ def history(self):
                     )
             except SQLError as e:
                 self.target.log.warning("Error processing history file: %s", db_file, exc_info=e)
+
+    @export(record=BROWSER_DOWNLOAD_RECORD)
+    def downloads(self) -> Iterator[BROWSER_DOWNLOAD_RECORD]:
+        """Return browser download records from Firefox.
+
+        Yields FirefoxBrowserHistoryRecord with the following fields:
+            hostname (string): The target hostname.
+            domain (string): The target domain.
+            ts_start (datetime): .
+            ts_end (datetime): .
+            browser (string): The browser from which the records are generated from.
+            id (string): Record ID.
+            path (string): .
+            url (uri): History URL.
+            size (string): .
+            state (string): .
+            source: (path): The source file of the history record.
+        """
+        for user, db_file, db in self._iter_db("places.sqlite"):
+            try:
+                # TODO: should implement proper indexing in SQLite at some point
+                places = {row.id: row for row in db.table("moz_places").rows()}
+                attributes = {row.id: row.name for row in db.table("moz_anno_attributes").rows()}
+                annotations = {}
+
+                for row in db.table("moz_annos"):
+                    attribute_name = attributes.get(row.anno_attribute_id, row.anno_attribute_id)
+
+                    if attribute_name == "downloads/metaData":
+                        content = json.loads(row.content)
+                    else:
+                        content = row.content
+
+                    if row.place_id not in annotations:
+                        annotations[row.place_id] = {"id": row.id}
+
+                    annotations[row.place_id][attribute_name] = {
+                        "content": content,
+                        "flags": row.flags,
+                        "expiration": row.expiration,
+                        "type": row.type,
+                        "date_added": from_unix_us(row.dateAdded),
+                        "last_modified": from_unix_us(row.lastModified),
+                    }
+
+                for place_id, annotation in annotations.items():
+                    if "downloads/metaData" not in annotation:
+                        continue
+
+                    place: Row = places[place_id]
```

Comment:
```suggestion
                    place = places[place_id]
```
We generally don't do type hinting here.